### ISSUE TITLE
Seed is dumped in trace

### ISSUE THREAD

**Issue by Techworker** [iota-php/client#25]
The seed shouldn't be added in clear text in traces, e.g.

https://github.com/Techworker/iota-php/blob/master/src/ClientApi/Actions/GetAddresses/Action.php#L201

Or this should be made configurable with a debug flag?

**Comment by peter279k** [comment]
I think it's not necessary to dump the `security` serialization by default in trace.

Perhaps we can set the optional argument in the `serialize` method to determine the security will be dumped in trace. Consider the following code snippets:

```php
public function serialize(bool isDumpedSecurity = false): array
{
    ...
}
```

**Comment by Techworker** [comment]
Well it implements the [SerializeInterface](https://github.com/iota-php/client/blob/master/src/SerializeInterface.php) and this method does not define this parameter.

**Comment by peter279k** [comment]
Ok. Is it possible to add the optional argument for the `serialize` method in `SerializeInterface`?

**Comment by Techworker** [comment]
This would change all classes that implement the SerializeInterface (a lot!!), and not all really need the flag.  

Btw., the line that is meant is this: https://github.com/iota-php/client/blob/master/src/ClientApi/Actions/GetAddresses/Action.php#L199

The [Seed](https://github.com/iota-php/client/blob/master/src/Type/Seed.php) class has some mechanisms to never get dumped anywhere, because the seed should never NEVER EVER appear in any logs or anywhere else. 

So in essence a better fix might be to overwrite the serialize() function from the Trytes class the Seed class extends from.

The whole serialization stuff is mostly for debugging purposes and its really helpful (see the [kitchen sink](https://github.com/iota-php/examples) which lets you try out each endpoint and client method via browser)

**Comment by peter279k** [comment]
Ok, but we have to find another way to let the `seed` be configurable and it checks whether this should be dumped in trace.

My another approach is: Let the seed be removed in this `serialize` method and I think it's the proper way to do that because the `seed` is the credentials.

**Comment by Techworker** [comment]
Yes, probably remove the serialization for the seed completely. More secure

**Comment by peter279k** [comment]
@Techworker, I've created the related PR and this issue should  be fixed :smiley: .

**Comment by Techworker** [comment]
Thanks, I just saw that there are more files where the seed is dumped. Do you have time to fix them all?

If so, search for "'seed' =>" in the ClientApi directory. I think we can remove them all.

**Comment by peter279k** [comment]
@Techworker, thank you for your reply.

I can help you for this request changes :smile:.

**Comment by Techworker** [comment]
Fixed with #55
